mlflow/models/cli.py

```diff
@@ -155,6 +155,35 @@ def prepare_env(
     ).prepare_env(model_uri=model_uri)
 
 
+@commands.command("generate-dockerfile")
+@cli_args.MODEL_URI_BUILD_DOCKER
+@click.option("--output-directory",
+              "-d",
+              default="generate-dockerfile-output",
+              help="output directory for generating dockerfile")
```

Comment:
Can we accept a path rather than a directory, e.g. `/tmp/my/custom/path/Dockerfile` and default to `Dockerfile` so that the default behavior produces a dockerfile in the current working directory?

Reply:
+1 for using a path. Can we validate the specified Dockerfile path doesn't exist?

Reply:
I started with a path, but encountered problems with `model_dir`. At first, I thought, that function will generate a `Dockerfile` only, but since `model_dir` used to copy model into image --- it should be present also.
This led me to believe that even if I pass a desired path to `Dockerfile` it will create a `model_dir ` in a parent directory, which seems to me as a little unexpected\not obvious, which can result in confusion for user --- since it only accepts dockerfile path, user can assume that only dockerfile is needed to build an image and can move it somewhere else *without moving `model_dir`*. 
Maybe I'm a little too cautious here, but folder seems as not ideal, but at least an obvious option

As for validating if a path doesn't exist --- function ensures creation of a path with `os.makedirs`, which allows us to pass any path

I am open to discussions and suggestions --- maybe I am just being too cautious here?

Reply:
@anuarkaliyev23 This is a great point. I think that producing a directory is the best option given that wrinkle. Can we rename the default directory to `mlflow-dockerfile` and update the helpstring to indicate that the directory contains a `Dockerfile` and also the resolved model, if `model-uri` is specified?